index.php

```diff
@@ -213,12 +213,45 @@ function gutenberg_scripts_and_styles( $hook ) {
 		'wp-editor',
 		plugins_url( 'editor/build/style.css', __FILE__ ),
 		array( 'wp-blocks' ),
-		filemtime( plugin_dir_path( __FILE__ ) . 'editor/build/style.css' )
+		gutenberg_asset_cache_version( 'editor/build/style.css' )
 	);
 }
 add_action( 'admin_enqueue_scripts', 'gutenberg_scripts_and_styles' );
 
 /**
+ * Set's a script/style version for cache busting.
+ *
+ * 0.1.0
+ *
+ * @param string $file The file path relative to the plugin's root.
+ *
+ * @return string
+ */
+function gutenberg_asset_cache_version( $file = null ) {
+	$version = false;
+
+	if ( defined( 'WP_DEBUG' ) && WP_DEBUG && $file ) {
+		if ( file_exists( plugin_dir_path( __FILE__ ) . $file ) ) {
+			$version = filemtime( plugin_dir_path( __FILE__ ) . $file );
+		}
+	} else {
+		$version = get_transient( 'gutenberg_version' );
+
+		if ( ! $version && function_exists( 'get_plugin_data' ) ) {
+			$plugin = get_plugin_data( __FILE__, false );
+
+			if ( ! empty( $plugin['Version'] ) ) {
+				$version = $plugin['Version'];
+			}
+
+			set_transient( 'gutenberg_version', $version, DAY_IN_SECONDS );
```

Comment:
I agree the second option is probably best and being more explicit by using `false === $version` is a good idea; however, allowing an empty value (while it may never be empty in this case) would not be ideal if we want the transient to update if the value were to return not empty (after set as empty).

How about something like:

```php
if ( false === $version && function_exists( 'get_plugin_data' ) ) {
	$plugin = get_plugin_data( __FILE__, false );

	if ( ! empty( $plugin['Version'] ) ) {
		$version = $plugin['Version'];

		set_transient( 'gutenberg_version', $version, DAY_IN_SECONDS );
	}
}
```